src/Hooks.tsx

```diff
@@ -23,7 +27,9 @@ function useBrowserStorage<T>(key: string, initialValue: T, type: 'session' | 'l
     try {
       const valueToStore = value instanceof Function ? value(storedValue) : value
       setStoredValue(valueToStore)
-      storage.setItem(key, JSON.stringify(valueToStore))
+
+      const storageString = isString(valueToStore) ? (valueToStore as string) : JSON.stringify(valueToStore);
```

Comment:
With `value is string` above, we won't need `as string` here.
```suggestion
      const storageString = isString(valueToStore) ? valueToStore : JSON.stringify(valueToStore);
```